src/map/entities/battleentity.cpp

```diff
@@ -226,17 +226,44 @@ int32 CBattleEntity::GetMaxMP() const
 
 uint8 CBattleEntity::GetSpeed()
 {
+    uint8 baseSpeed   = speed;
+    uint8 outputSpeed = 0;
+
+    // Mount speed. Independent from regular speed and unaffected by most things.
     // Note: retail treats mounted speed as double what it actually is! 40 is in fact retail accurate!
-    int16 startingSpeed = isMounted() ? 40 + settings::get<int8>("map.MOUNT_SPEED_MOD") : speed;
-    // Mod::MOVE (169)
-    // Mod::MOUNT_MOVE (972)
-    Mod mod = isMounted() ? Mod::MOUNT_MOVE : Mod::MOVE;
+    if (isMounted())
+    {
+        baseSpeed   = 40 + settings::get<int8>("map.MOUNT_SPEED_MOD");
+        outputSpeed = baseSpeed * (100 + getMod(Mod::MOUNT_MOVE)) / 100;
+
+        return std::clamp<uint8>(outputSpeed, 0, 80);
```

Comment:
please don't clamp final chocobo/mount to 80. We want the setting to allow ridiculous chocobo speeds. It won't get that high without someone intentionally putting the setting way up there.